app/controllers/works_controller.rb

```diff
@@ -0,0 +1,69 @@
+class WorksController < ApplicationController
+
+  def index
+    @books = Work.sort_by_votes('book')
+    @albums = Work.sort_by_votes('album')
+    @movies = Work.sort_by_votes('movie')
+  end
+
+  def show
+    @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      head :not_found
+    end
```

Comment:
This sort of repeated logic should be placed into a [controller filter](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/09-intermediate-rails/controller-filters.md).